addon/-private/system/record-arrays/record-array.js

```diff
@@ -201,9 +202,16 @@ export default Ember.ArrayProxy.extend(Ember.Evented, {
     this._super.apply(this, arguments);
   },
 
-  createSnapshot(options) {
-    var adapterOptions = options && options.adapterOptions;
-    var meta = this.get('meta');
-    return new SnapshotRecordArray(this, meta, adapterOptions);
+  createSnapshot(options = {}) {
+    const adapterOptions = options.adapterOptions;
+    const meta = this.get('meta');
+
+    const snapshot = new SnapshotRecordArray(this, meta, adapterOptions);
```

Comment:
it might be nice if the two snapshot constructors had the same support for providing `adapterOptions` and possibly other options. It seems a little weird that we set properties on it directly after we initialize it rather than using the constructor. Thoughts?